apps/prairielearn/src/pages/studentAssessment/studentAssessment.ts

```diff
@@ -150,6 +150,10 @@ router.post(
       });
       res.redirect(`${res.locals.urlPrefix}/assessment_instance/${assessment_instance_id}`);
     } else if (req.body.__action === 'join_group') {
+      const groupConfig = await getGroupConfig(res.locals.assessment.id);
+      if (!groupConfig.student_authz_join) {
+        throw new HttpStatusError(403, 'You are not authorized to join a group.');
+      }
```

Comment:
👀 were we just straight up missing validation before?

Reply:
Indeed, only client-side validation (via hiding buttons/forms) was in place.